lib/modules/datasource/packagist/index.ts

```diff
@@ -89,7 +85,7 @@ export class PackagistDatasource extends Datasource {
 
   private static isPrivatePackage(regUrl: string): boolean {
     const opts = PackagistDatasource.getHostOpts(regUrl);
-    return !!opts.password || !!opts.headers?.authorization;
+    return !!opts.password;
```

Comment:
What if authorization header is applied due to hostRules by the user, e.g. for a private registry?

Reply:
We probably should reflect this in `getHostOpts()` logic, currently there is no way for `opts` to have `headers.authorization` field. Do you think we need to provide `headers` in the `getHostOpts()` result?

Reply:
Ok good point. I was assuming it was based on the res.headers.

It should match, whatever we do